src/firefish/firefish.rs

```diff
@@ -2429,7 +2429,20 @@ impl megalodon::Megalodon for Firefish {
         ))
     }
 
-    fn user_streaming(&self, streaming_url: String) -> Box<dyn Streaming + Send + Sync> {
+    async fn streaming_url(&self) -> String {
+        let instance = self.get_instance().await;
+        if let Ok(instance) = instance {
+            match instance.json.urls {
+                Some(urls) => return urls.streaming_api,
+                _ => {}
+            };
+        }
+
+        self.base_url.clone()
+    }
+
+    async fn user_streaming(&self) -> Box<dyn Streaming + Send + Sync> {
+        let streaming_url = self.streaming_url().await;
         let c = WebSocket::new(
             streaming_url + "/streaming",
```

Comment:
It looks like Firefish in particular may just need to use `streaming_url` verbatim? https://calckey.world/api/v1/instance